core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -525,14 +528,20 @@ private void catchUp(JournalStateMachine stateMachine, RaftClient client)
       long gainPrimacySN = ThreadLocalRandom.current().nextLong(Long.MIN_VALUE, 0);
       LOG.info("Performing catchup. Last applied SN: {}. Catchup ID: {}",
           lastAppliedSN, gainPrimacySN);
-      CompletableFuture<RaftClientReply> future = client.sendAsync(
-          toRaftMessage(JournalEntry.newBuilder().setSequenceNumber(gainPrimacySN).build()));
+      Exception ex;
       try {
-        future.get(5, TimeUnit.SECONDS);
-      } catch (TimeoutException | ExecutionException e) {
-        client.getClientRpc().handleException(mPeerId,
-            e instanceof ExecutionException ? e.getCause() : e, true);
-        LOG.info("Exception submitting term start entry: {}", e.toString());
+        CompletableFuture<RaftClientReply> future = client.sendAsync(
+            toRaftMessage(JournalEntry.newBuilder().setSequenceNumber(gainPrimacySN).build()),
+            TimeDuration.valueOf(5, TimeUnit.SECONDS));
+        RaftClientReply reply = future.get(5, TimeUnit.SECONDS);
+        ex = reply.getException();
+      } catch (TimeoutException | ExecutionException | IOException e) {
+        ex = e;
+      }
+      if (ex != null) {
+        LOG.info("Exception submitting term start entry: {}", ex.toString());
+        // avoid excessive retries when server is not ready
+        Thread.sleep(100);
```

Comment:
Should this be configurable? I can't tell if 100ms is too short or too long. This means we will retry 10x per second forever? That seems quite frequent.

Reply:
Yeah it is still significantly better than retrying immediately and blowing up the logs in no time. I added a configuration property for 1s.